### ISSUE TITLE
[Bug] Mismatched versions for Python 3.11 on torch

### ISSUE THREAD

**Issue by simwijs-evolante** [pinecone-io/pinecone-text#32]
### Is this a new bug?

- [X] I believe this is a new bug
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Installing using pipenv fails  due to
```
The conflict is caused by:
    pinecone-text 0.4.2 depends on torch<2.0.0 and >=1.13.1
    sentence-transformers 2.0.0 depends on torch>=1.6.0
    torchvision 0.15.2 depends on torch==2.0.1
    pinecone-text 0.4.2 depends on torch<2.0.0 and >=1.13.1
    sentence-transformers 2.0.0 depends on torch>=1.6.0
    torchvision 0.15.1 depends on torch==2.0.0
```

### Expected Behavior

No conflicting versions

### Steps To Reproduce

Use this Pipfile
```
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
pinecone-text = "==0.4.2"
[dev-packages]

[requires]
python_version = "3.11"

```

and this Dockerfile
```
# python 3.11
FROM python:bullseye as base  

# Setup env
ENV LANG C.UTF-8
ENV LC_ALL C.UTF-8
ENV PYTHONDONTWRITEBYTECODE 1
ENV PYTHONFAULTHANDLER 1

FROM base AS python-deps

# Install pipenv and compilation dependencies
RUN pip install pipenv
RUN apt-get update && apt-get install -y --no-install-recommends gcc

# Install python dependencies in /.venv
COPY Pipfile .
# COPY Pipfile.lock .
RUN PIPENV_VENV_IN_PROJECT=1 pipenv install

FROM base AS runtime

# Copy virtual env from python-deps stage
COPY --from=python-deps /.venv /.venv
ENV PATH="/.venv/bin:$PATH"
```

### Relevant log output

```shell
Installing pinecone-text==0.4.2...
Resolving pinecone-text==0.4.2...
Installing...
⠋ Installing pinecone-text...[31m[1mError: [0m An error occurred while installing [32mpinecone-text==0.4.2[0m!
Error text: Collecting pinecone-text==0.4.2
  Using cached pinecone_text-0.4.2-py3-none-any.whl (17 kB)
Collecting mmh3<4.0.0,>=3.1.0
  Using cached mmh3-3.1.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (38 kB)
Collecting nltk<4.0.0,>=3.6.5
  Using cached nltk-3.8.1-py3-none-any.whl (1.5 MB)
Collecting sentence-transformers<3.0.0,>=2.0.0
  Using cached sentence-transformers-2.2.2.tar.gz (85 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting torch<2.0.0,>=1.13.1
  Using cached torch-1.13.1-cp311-cp311-manylinux1_x86_64.whl (887.4 MB)
Collecting transformers<5.0.0,>=4.26.1
  Using cached transformers-4.28.1-py3-none-any.whl (7.0 MB)
Collecting wget<4.0,>=3.2
  Using cached wget-3.2.zip (10 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting joblib
  Using cached joblib-1.2.0-py3-none-any.whl (297 kB)
Collecting regex>=2021.8.3
  Using cached regex-2023.5.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (780 kB)
Collecting tqdm
  Using cached tqdm-4.65.0-py3-none-any.whl (77 kB)
Collecting torchvision
  Using cached torchvision-0.15.2-cp311-cp311-manylinux1_x86_64.whl (6.0 MB)
Requirement already satisfied: numpy in /home/sws/.local/lib/python3.11/site-packages (from sentence-transformers<3.0.0,>=2.0.0->pinecone-text==0.4.2->-r /tmp/pipenv-wrvm0n__-requirements/pipenv-haacb93p-requirement.txt (line 1)) (1.24.2)
Collecting scikit-learn
  Using cached scikit_learn-1.2.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (9.6 MB)
Collecting scipy
  Using cached scipy-1.10.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.1 MB)
Collecting sentencepiece
  Using cached sentencepiece-0.1.99-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)
Collecting huggingface-hub>=0.4.0
  Using cached huggingface_hub-0.14.1-py3-none-any.whl (224 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Collecting nvidia-cuda-runtime-cu11==11.7.99
  Using cached nvidia_cuda_runtime_cu11-11.7.99-py3-none-manylinux1_x86_64.whl (849 kB)
Collecting nvidia-cudnn-cu11==8.5.0.96
  Using cached nvidia_cudnn_cu11-8.5.0.96-2-py3-none-manylinux1_x86_64.whl (557.1 MB)
Collecting nvidia-cublas-cu11==11.10.3.66
  Using cached nvidia_cublas_cu11-11.10.3.66-py3-none-manylinux1_x86_64.whl (317.1 MB)
Collecting nvidia-cuda-nvrtc-cu11==11.7.99
  Using cached nvidia_cuda_nvrtc_cu11-11.7.99-2-py3-none-manylinux1_x86_64.whl (21.0 MB)
Requirement already satisfied: setuptools in /home/sws/.local/lib/python3.11/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch<2.0.0,>=1.13.1->pinecone-text==0.4.2->-r /tmp/pipenv-wrvm0n__-requirements/pipenv-haacb93p-requirement.txt (line 1)) (67.4.0)
Requirement already satisfied: wheel in /home/sws/.local/share/virtualenvs/pinecone-fastapi-Be-Gh9Zv/lib/python3.11/site-packages (from nvidia-cublas-cu11==11.10.3.66->torch<2.0.0,>=1.13.1->pinecone-text==0.4.2->-r /tmp/pipenv-wrvm0n__-requirements/pipenv-haacb93p-requirement.txt (line 1)) (0.40.0)
Requirement already satisfied: filelock in /home/sws/.local/lib/python3.11/site-packages (from transformers<5.0.0,>=4.26.1->pinecone-text==0.4.2->-r /tmp/pipenv-wrvm0n__-requirements/pipenv-haacb93p-requirement.txt (line 1)) (3.9.0)
Collecting packaging>=20.0
  Using cached packaging-23.1-py3-none-any.whl (48 kB)
Collecting pyyaml>=5.1
  Using cached PyYAML-6.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (757 kB)
Collecting requests
  Using cached requests-2.30.0-py3-none-any.whl (62 kB)
Collecting tokenizers!=0.11.3,<0.14,>=0.11.1
  Using cached tokenizers-0.13.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (7.8 MB)
Collecting fsspec
  Using cached fsspec-2023.5.0-py3-none-any.whl (160 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.1.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (197 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<3,>=1.21.1
  Using cached urllib3-2.0.2-py3-none-any.whl (123 kB)
Requirement already satisfied: certifi>=2017.4.17 in /home/sws/.local/lib/python3.11/site-packages (from requests->transformers<5.0.0,>=4.26.1->pinecone-text==0.4.2->-r /tmp/pipenv-wrvm0n__-requirements/pipenv-haacb93p-requirement.txt (line 1)) (2022.12.7)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting torchvision
  Using cached torchvision-0.15.1-cp311-cp311-manylinux1_x86_64.whl (6.0 MB)
INFO: pip is looking at multiple versions of sentencepiece to determine which version is compatible with other requirements. This could take a while.
Collecting sentencepiece
  Using cached sentencepiece-0.1.98-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)
INFO: pip is looking at multiple versions of scipy to determine which version is compatible with other requirements. This could take a while.
Collecting scipy
  Using cached scipy-1.10.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.1 MB)
INFO: pip is looking at multiple versions of scikit-learn to determine which version is compatible with other requirements. This could take a while.
Collecting scikit-learn
  Using cached scikit_learn-1.2.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (9.6 MB)
INFO: pip is looking at multiple versions of requests to determine which version is compatible with other requirements. This could take a while.
Collecting requests
  Using cached requests-2.29.0-py3-none-any.whl (62 kB)
INFO: pip is looking at multiple versions of joblib to determine which version is compatible with other requirements. This could take a while.
Collecting joblib
  Using cached joblib-1.1.1-py2.py3-none-any.whl (309 kB)
INFO: pip is looking at multiple versions of filelock to determine which version is compatible with other requirements. This could take a while.
Collecting filelock
  Using cached filelock-3.12.0-py3-none-any.whl (10 kB)
INFO: pip is looking at multiple versions of click to determine which version is compatible with other requirements. This could take a while.
Collecting click
  Using cached click-8.1.2-py3-none-any.whl (96 kB)
INFO: pip is looking at multiple versions of typing-extensions to determine which version is compatible with other requirements. This could take a while.
Collecting typing-extensions
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
INFO: pip is looking at multiple versions of tqdm to determine which version is compatible with other requirements. This could take a while.
Collecting tqdm
  Using cached tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
INFO: pip is looking at multiple versions of tokenizers to determine which version is compatible with other requirements. This could take a while.
Collecting tokenizers!=0.11.3,<0.14,>=0.11.1
  Using cached tokenizers-0.13.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (7.6 MB)
INFO: pip is looking at multiple versions of regex to determine which version is compatible with other requirements. This could take a while.
Collecting regex>=2021.8.3
  Using cached regex-2023.5.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (780 kB)
INFO: pip is looking at multiple versions of pyyaml to determine which version is compatible with other requirements. This could take a while.
Collecting pyyaml>=5.1
  Using cached PyYAML-5.4.1.tar.gz (175 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
INFO: pip is looking at multiple versions of packaging to determine which version is compatible with other requirements. This could take a while.
Collecting packaging>=20.0
  Using cached packaging-23.0-py3-none-any.whl (42 kB)
INFO: pip is looking at multiple versions of numpy to determine which version is compatible with other requirements. This could take a while.
Collecting numpy
  Using cached numpy-1.24.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
INFO: pip is looking at multiple versions of huggingface-hub to determine which version is compatible with other requirements. This could take a while.
Collecting huggingface-hub>=0.4.0
  Using cached huggingface_hub-0.14.0-py3-none-any.whl (224 kB)
INFO: pip is looking at multiple versions of wget to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of transformers to determine which version is compatible with other requirements. This could take a while.
Collecting transformers<5.0.0,>=4.26.1
  Using cached transformers-4.28.0-py3-none-any.whl (7.0 MB)
INFO: pip is looking at multiple versions of nvidia-cudnn-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of nvidia-cuda-runtime-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of nvidia-cuda-nvrtc-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of nvidia-cublas-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of torch to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of sentence-transformers to determine which version is compatible with other requirements. This could take a while.
Collecting sentence-transformers<3.0.0,>=2.0.0
  Using cached sentence-transformers-2.2.1.tar.gz (84 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Using cached sentence-transformers-2.2.0.tar.gz (79 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Using cached sentence-transformers-2.1.0.tar.gz (78 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Using cached sentence-transformers-2.0.0.tar.gz (85 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
INFO: pip is looking at multiple versions of nltk to determine which version is compatible with other requirements. This could take a while.
Collecting nltk<4.0.0,>=3.6.5
  Using cached nltk-3.8-py3-none-any.whl (1.5 MB)
INFO: pip is looking at multiple versions of nvidia-cudnn-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of nvidia-cuda-runtime-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of nvidia-cuda-nvrtc-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of nvidia-cublas-cu11 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of torch to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of sentence-transformers to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of mmh3 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of pinecone-text to determine which version is compatible with other requirements. This could take a while.

The conflict is caused by:
    pinecone-text 0.4.2 depends on torch<2.0.0 and >=1.13.1
    sentence-transformers 2.0.0 depends on torch>=1.6.0
    torchvision 0.15.2 depends on torch==2.0.1
    pinecone-text 0.4.2 depends on torch<2.0.0 and >=1.13.1
    sentence-transformers 2.0.0 depends on torch>=1.6.0
    torchvision 0.15.1 depends on torch==2.0.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

### Environment

```markdown
- **OS**: See Dockerfile
- **Language version**: -
- **Pinecone client version**: -
```

### Additional Context

none

**Comment by btconway** [comment]
Also encountering this same issue.

**Comment by FrackinFamous** [comment]
I'm not sure if you guys have solved this but I would do as it recommends and delete all the version numbers from your requirements.txt and let pip sort it. 

If not delete your deps and reinstall. 

Im assuming it stems from the dockerfile copying the venv that was used on the original system. You should not copy a venv just the requirements.txt. Since they copied the whole venv they may be getting weird dep issues due to dependencies of the OG machine that the Dockerfile was build from or from conflicts with the machine you are currently running the Dockerfile on.  

This is why when you Git clone. You don't get the venv with the clone.

**Comment by acatav** [comment]
Thanks for pointing it out and suggest PRs, the issue is now fixed at version `0.5.1`
